### ISSUE TITLE
Jakarta EE 10: Support Jakarta Authorization 2.1

### ISSUE THREAD

**Issue by jhanders34** [OpenLiberty/open-liberty#18726]
## Description of the high level feature, including any external spec links:  
Implement the Jakarta Authorization 2.1 specification

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## When complete & mandatory, add links to the UFO (Upcoming Feature Overview) document, FTS (Feature Test Summary), and blogs post issues(s):
- UFO:  https://ibm.ent.box.com/file/953288710429
- FTS: https://github.com/OpenLiberty/open-liberty/issues/23023
- Demo: scheduled for Jan 31, 2023
- Beta Blog Post (if applicable):https://github.com/OpenLiberty/open-liberty/issues/23355
- Blog Post: 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [x] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [x] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [ ] No Design / No UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified ([Documenting Open Liberty](https://github.com/OpenLiberty/open-liberty/wiki/Documenting-Open-Liberty)). (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of this epic entitled "Feature Test Summary" via [this template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=Feature+Test+Summary&template=feature_test_summary.md&title=). Add the link in above.

##
##

## **Development**
When active development has begun 
- [x] Add the "In Progress" label to this issue.

## **Beta**
If your feature, or portions of it, are going to be included in a beta  
Before Onboarding the beta
- [x] Beta Fence the functionality (`kind=beta`, `ibm:beta`, `ProductInfo.getBetaEdition()`)  

1 week before beta GA
- [x] Create, populate, and link to the [Beta blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_beta.md&title=BETA+BLOG+-+title_of_your_update)

## **Legal**
3 weeks before Onboarding
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [x] **FAT** - ([OpenLiberty/fat-approvers](https://github.com/orgs/OpenLiberty/teams/fat-approvers)). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Demo** - ([Tom Evans](https://github.com/tevans78) or [Chuck Bridgham](https://github.com/cbridgha)). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [x] **Globalization** ([Sam Wong](https://github.com/samwatibm) - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [x] **Accessibility** - ([Steven Zvonek](https://github.com/steven1046)). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **ID** - ([Karen Deen](https://github.com/chirp1)). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - ([Jared Anderson](https://github.com/jhanders34)). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - ([Don Bourne](https://github.com/donbourne)). Serviceability has been addressed.
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - ([Brian Hanczaryk](https://github.com/hanczaryk) - APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the release branch. (Epic owner / Feature owner / Backlog Subtribe PM)

1 week before GA
- [ ] Create, populate, and link to the [Blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_ga_release.md&title=GA+BLOG+-+title_of_your_update)

## **Other deliverbles**
- [x] **OL Guides** - ([Yee-Kang Chang](https://github.com/yeekangc)). Assessment for OL Guides is complete or N/A.
- [x] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [x] **Blog** - ([Laura Cowen](https://github.com/lauracowen)) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by emilytee** [comment]
Link to UFO: https://ibm.ent.box.com/file/953288710429
Link to FTS issue:  https://github.com/OpenLiberty/open-liberty/issues/23023

**Comment by emilytee** [comment]
@samw - no new translation/messages were dropped as part of this feature.  Please update the feature' globalization approval.

**Comment by emilytee** [comment]
UPDATED UFO LINK:  https://ibm.ent.box.com/file/973577297689?s=zrabsaicm2bkfsywft6v6j366u8jeezd

**Comment by chirp1** [comment]
Emily approved the doc updates for this issue at https://github.com/OpenLiberty/docs/issues/5851 . Approving this feature.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

5.  Does this feature add any new metrics or emit any new JSON events?  If yes, have you updated the JMX metrics reference list / Metrics reference list / JSON log events reference list in the Open Liberty docs?

**Comment by emilytee** [comment]
Serviceability Approval Comment - Please answer the following questions for serviceability approval:

UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?   The UFO identifies the Jakarta spec method additions to the existing authorization object which are validated both with FVT as well as CTS/TCK testing to ensure spec compliance.   All issues are addressed via the implementation of the spec      
    
Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).   
a) What problem paths were tested and demonstrated?   All identified failing and positive paths were reviewed by the test team not involved with the developmet of the feature
b) Who did you demo to?   test team
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3? yes

SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?   It was agreed with the SVT team that no SVT was needed 
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?   n/a

Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info.    L2/L3 Security.   They have been involved in the reviews.

Does this feature add any new metrics or emit any new JSON events? If yes, have you updated the JMX metrics reference list / Metrics reference list / JSON log events reference list in the Open Liberty docs?   No new metrics/JSON events.

**Comment by donbourne** [comment]
see https://github.com/OpenLiberty/open-liberty/issues/18725#issuecomment-1332582157

**Comment by emilytee** [comment]
Link to STE:  https://ibm.ent.box.com/file/1080317751188  
@gnadell @tngiang73

**Comment by gnadell** [comment]
@emilytee ,WASSEC L2 is good with the STE slides. Hence approving.

**Comment by malincoln** [comment]
@emilytee pls schedule the demo (if not already scheduled) and add the date of demo for approval.  Thanks!

**Comment by emilytee** [comment]
@hanczaryk , can you please add your SVT approval.   This work item is a part of the overall Security Jakarta EE10 UFO, where it  was decided no specific SVT tests would be run other than the existing set.

**Comment by hanczaryk** [comment]
@emilytee SVT still has outstanding regression tests for several of our SVT applications. SVT doesn't mark focal point approval until all SVT tests have completed. Also, we'll run these regression tests again on builds closer to GM, but as of now a few applications (QuickSec, ACME, ERWW) haven't completed a successful regression run using Jakarta EE 10. I will mark approval when those SVT stress tests complete.

**Comment by malincoln** [comment]
@tevans78 @cbridgha not sure if you saw where Emily added the demo dates at top but adding here:
Demo: scheduled for Jan 31, 2023
